### PR TITLE
Send different email updates to different types of ticket holders, fix #283

### LINKED ISSUE #283: Send different types of email updates to different types of ticket holders
Add options for sending different emails to different types of ticket holders. To increase character limit and Rich Text. For both Admin and Host users.

### PR DESCRIPTION
**Pull request by pro-Nate**

### CODE REVIEW

File: py/web/views/events.py [hunk 1]
```diff
@@ -872,19 +872,25 @@ async def execute(self, m: Model):
...
         subject: constr(max_length=200)
-        message: str
+        message: Optional[str]
 
     async def check_permissions(self):
         await check_session(self.request, 'admin', 'host')
 
     async def execute(self, m: Model):
         event_id = await _check_event_permissions(self.request, check_upcoming=True)
+        payload_json = await self.request.json()
```
Comment (samuelcolvin) [inline py/web/views/events.py:882]: doesn't really make sense to use the json payload when you have `Model`

File: py/web/views/events.py [hunk 2]
```diff
@@ -872,19 +872,25 @@ async def execute(self, m: Model):
...
+        msgs = {k: v for k, v in payload_json.items() if (k.startswith('message'))}
+
+        if not len(msgs):
```
Comment (samuelcolvin) [inline py/web/views/events.py:885]:
```suggestion
        if not msgs:
```
this is python, not js :-)

**Review by samuelcolvin — COMMENTED** [review]
a few details to change, but I think better to make a more general change:

Instead of a different body for each ticket type, better to have a dropdown for ticket type on the event update form, and just one subject and message box.

Then if you want to sent an email to each ticket type, you submit the form multiple times. You could even make ticket type optional, incase they want to send to everyone.

I think this would better reflect how people will use it, and mean fewer code changes.

File: py/web/views/events.py (hunk 2 above)
Comment (pro-Nate) [reply py/web/views/events.py]: lol thanks

File: py/web/views/events.py (hunk 1 above)
Comment (pro-Nate) [reply py/web/views/events.py:882]: Ah yes. The view doesn't know what the possible id's that will receive messages. I haven't mastered pydantic yet and wasn't sure how to define a field with a wildcard or some other way to include all messages into `m`.

**Comment by pro-Nate** [comment]
> Instead of a different body for each ticket type, better to have a dropdown for ticket type on the event update form, and just one subject and message box.

I can propose that to handsup, but at present I've implemented it the way they described in meeting.

**Comment by samuelcolvin** [comment]
> > Instead of a different body for each ticket type, better to have a dropdown for ticket type on the event update form, and just one subject and message box.
> 
> I can propose that to handsup, but at present I've implemented it the way they described in meeting.

I would strongly suggest to them that they do it this way. It'll be much easier to use, and more flexible - e.g. if you only want to email one group, the current solution wouldn't work.

**Comment by pro-Nate** [comment]
Amira has asked to try this solution before deciding.

**Comment by samuelcolvin** [comment]
I absolutely disagree with merging code like this.

If we start doing things like

```py
payload_json = await self.request.json()
```

And intentionally avoiding validation, but keeping the validation in the codebase in order to confuse a future developers (it's not doing anything else), the codebase will be unusable in a matter of months.

You don't need to merge PRs to deploy them for testing.

**This is a route to errors and getting hacked.**

@pro-Nate please revert this pull request.

**Pull request merged**